nodejs/week1/src/backend/index.js

```diff
@@ -0,0 +1,85 @@
+const express = require("express");
+
+const app = express();
+const port = 3000;
+
+///meals///
+/*app.get("/meals", (request, response) => {
+  let mealsData = getDataFromJson(meals);
+  let reviewsData = getDataFromJson(reviews);
+  let mealsWithReviews = mealsData.map((meal) => {
+    meal.review = reviewsData.filter((review) => review.id === meal.id);
+    return meal;
+  });
+  response.send(mealsWithReviews);
+});*/
+
+const mealsRouter = require("./routes/meals.js");
+app.get("/meals", mealsRouter);
+
+////cheap-meals///
+/*app.get("/cheap-meals", (request, response) => {
+  let mealsData = getDataFromJson(meals);
+  let reviewsData = getDataFromJson(reviews);
+  let mealsWithReviews = mealsData.map((meal) => {
+    meal.review = reviewsData.filter((review) => review.id === meal.id);
+    return meal;
+  });
+  let cheapMealsWithReviews = mealsWithReviews.filter((meal) => meal.price < 70)
+  response.send(cheapMealsWithReviews);
+});*/
+
+const cheapMealRouter = require("./routes/cheap-meals.js");
+app.get("/cheap-meals", cheapMealRouter);
+
+///large-meals///
+/*app.get("/large-meals", (request, response) => {
+  let mealsData = getDataFromJson(meals);
+  let reviewsData = getDataFromJson(reviews);
+  let mealsWithReviews = mealsData.map((meal) => {
+    meal.review = reviewsData.filter((review) => review.id === meal.id);
+    return meal;
+  });
+  let largeMealsWithReviews = mealsWithReviews.filter((meal) => meal.maxNumberOfGuests > 6);
+  response.send(largeMealsWithReviews);
+});*/
+
+const largeMealRouter = require("./routes/large-meals.js");
+app.get("/large-meals", largeMealRouter);
+
+///meal///
+/*app.get("/meal", (request, response) => {
+  let mealsData = getDataFromJson(meals);
+  let reviewsData = getDataFromJson(reviews);
+  let mealsWithReviews = mealsData.map((meal) => {
+    meal.review = reviewsData.filter((review) => review.id === meal.id);
+    return meal;
+  });
+  let randomMeal = mealsWithReviews[Math.floor(Math.random() * mealsWithReviews.length)]
+  response.send(randomMeal);
+});*/
+const mealRouter = require("./routes/meal.js");
+app.get("/meal", mealRouter);
+
+///reservations///
+/*app.get("/reservations", (request, response) => {
+  let reservationsData = getDataFromJson(reservations)
+  response.send(reservationsData);
+});*/
+
+const reservationsRouter = require("./routes/reservations.js");
+app.get("/reservations", reservationsRouter);
+
+///reservation///
+/*app.get("/reservation", (request, response) => {
+  let reservationsData = getDataFromJson(reservations)
+  let randomReservationsData = reservationsData[Math.floor(Math.random() * reservationsData.length)]
+  response.send(randomReservationsData);
+});*/
+
+const reservationRouter = require("./routes/reservation.js");
+app.get("/reservation", reservationRouter);
```

Comment:
By convention we put all the imports (requires) at the top of the file. Otherwise it is hard to understand what is going on in a file because imports are intermingled with code. So use:

```
// ------- imports/requires ---------------
// these are all the things I need in this file that
// come from other files

// -------- actual code --------
// this is the purpose of this file. Why this file exists
// it is the most relevant part of the file. That is why
// it is specially important that it is super tidy and clear
```

A nice metaphor are ingredients and a recipe. The ingredients are what you need to make something work, and the recipe are the steps you follow to create a dish using the ingredients above. The imports/requires are your ingredients. The code that you write is your recipe. Ingredients first. Then use the ingredients in your recipe :)